pkg/controller/common/hash.go

```diff
@@ -0,0 +1,47 @@
+package common
+
+import (
+	//nolint:gosec
+	"crypto/md5"
+	"fmt"
+
+	"github.com/ghodss/yaml"
+	mcfgv1 "github.com/openshift/api/machineconfiguration/v1"
+)
+
+var (
+	// salt is 80 random bytes.
+	// The salt was generated by `od -vAn -N80 -tu1 < /dev/urandom`. Do not change it.
+	salt = []byte{
+		16, 124, 206, 228, 139, 56, 175, 175, 79, 229, 134, 118, 157, 154, 211, 110,
+		25, 93, 47, 253, 172, 106, 37, 7, 174, 13, 160, 185, 110, 17, 87, 52,
+		219, 131, 12, 206, 218, 141, 116, 135, 188, 181, 192, 151, 233, 62, 126, 165,
+		64, 83, 179, 119, 15, 168, 208, 197, 146, 107, 58, 227, 133, 188, 238, 26,
+		33, 26, 235, 202, 32, 173, 31, 234, 41, 144, 148, 79, 6, 206, 23, 22,
+	}
+)
+
+func HashMachineConfigSpec(spec mcfgv1.MachineConfigSpec) (string, error) {
```

Comment:
praise: Nicely done with this!

suggestion: Migrate rendercontroller to use this function instead of the one it's currently using. We can also migrate the MachineOSBuild function to use this too if we make it slightly more generic like this:

```golang
func GetHashForName(in interface{}) (string, error) {
	data, err := yaml.Marshal(in)
	if err != nil {
		return "", err
	}

	h, err := hashData(data)
	if err != nil {
		return "", err
	}
	return fmt.Sprintf("%x", h), nil
}
```